deploy-board/deploy_board/webapp/cluster_view.py

```diff
@@ -324,50 +324,65 @@ def parse_cluster_info(request):
     cluster_info['hostType'] = params['hostTypeId']
     cluster_info['securityZone'] = params['securityZoneId']
     cluster_info['placement'] = ",".join(params.getlist('placementId'))
+
+    # Update cluster name and isDocker in env
+    env_info = {}
+    env_info['clusterName'] = cluster_name
     if 'isDocker' in params:
-        cluster_info['isDocker'] = True
+        env_info['isDocker'] = True
     else:
-        cluster_info['isDocker'] = False
+        env_info['isDocker'] = False
+    environs_helper.update_env_basic_config(request, env_name, env_stage, data=env_info)
     return cluster_info
 
 
 def create_cmp_cluster(request, name, stage):
     params = request.POST
-    cluster_info = parse_cluster_info(request)
+    cluster_name = '{}-{}'.format(name, stage)
+    cluster_info = parse_cluster_info(request, name, stage, cluster_name)
     config_map = get_default_cmp_configs(name, stage)
     user_data_configs = parse_configs(params)
     config_map.update(user_data_configs)
     cluster_info['configs'] = config_map
-    clusters_helper.create_cluster(request, name, stage, cluster_info)
+    clusters_helper.create_cluster(request, cluster_name, cluster_info)
+
+    # set up env and group relationship
+    environs_helper.add_env_capacity(request, name, stage, capacity_type="GROUP", data=cluster_name)
     return get_cmp_cluster(request, name, stage)
 
 
 def create_cluster(request, name, stage):
```

Comment:
We might want to refactoring the create_cluster and create_cmp_cluster to combine them into one. can we add a TODO here to remind us?